runtime/shared_common/CacheMap.cpp

```diff
@@ -6327,13 +6327,10 @@ SH_CacheMap::matchAotMethod(MethodSpecTable* specTable, IDATA numSpecs, J9UTF8*
 		if (NULL == specTable[i].methodSig) {
 			/* Any signature */
 			sigMatch = true;
-		} else if (NULL == J9UTF8_DATA(romMethodSig)) {
-			/* Signature is missing */
-			sigMatch = false;
 		} else {
 			char* methodSig = (char *)J9UTF8_DATA(romMethodSig);
-			char* sigStart = strchr(methodSig, '(') + 1;
-			char* sigEnd = strchr(methodSig, ')');
+			char* sigStart = (char *)memchr(methodSig, '(', J9UTF8_LENGTH(romMethodSig)) + 1;
+			char* sigEnd = (char *)memchr(methodSig, ')', J9UTF8_LENGTH(romMethodSig));
```

Comment:
All three of the locals should have type `const char *`.
The cast on line 6342 was unnecessary and can be removed.
This should anticipate that `memchr()` may return `NULL`:
```c
			const char *methodSig = (const char *)J9UTF8_DATA(romMethodSig);
			const char *sigStart  = (const char *)memchr(methodSig, '(', J9UTF8_LENGTH(romMethodSig));
			const char *sigEnd    = (const char *)memchr(methodSig, ')', J9UTF8_LENGTH(romMethodSig));

			if ((NULL == sigStart) || (NULL == sigEnd) || (sigEnd <= sigStart)) {
				sigMatch = false;
			} else {
				/* Use string between '(' and ')' only. */
				IDATA sigLength = sigEnd - sigStart - 1;

				sigMatch = (TRUE == wildcardMatch(specTable[i].methodSigMatchFlag,
										(const char *)specTable[i].methodSig, specTable[i].methodSigLength,
										sigStart, sigLength));
			}
```

Reply:
Since we are in C++ code, we should use C++ casts instead of C-style casts. `static_cast<const char *>(...)` would be appropriate here since casting from `void *` to `T *` is well-defined.

Reply:
Can you cast `U_8 *` to `const char *` in the same way? The compiler complains when I try to use a C++ cast there, but a C cast seems fine.

Reply:
I'm not sure about introducing C++-style casts in that file (there aren't any before this change).
If we do, then `reinterpret_cast()` is needed to go from `U_8 *` to `char *` (`const` can be explicit, or implicit in the assignment).

BTW, when you wrote "casting from void * to T * is well-defined", did you mean "... **_not_** well-defined"?

Reply:
No, I meant well-defined. See https://en.cppreference.com/w/cpp/language/static_cast
> 11) A prvalue of type pointer to void (possibly cv-qualified) can be converted to pointer to any object type T.